### DISCUSSION TITLE
MatTableDataSource in Angular material

### DISCUSSION THREAD

**Discussion by rahulpd101** [ng-matero/ng-matero#587] [Q&A]
Hi,
I am trying to implement a filter functionality in MtxGridColumn using a class which has an inbuilt method to filter out required data. The issue I am having is I am not able to find any such class like 'MatTableDataSource' through which I can initialize my datalist which is accepted by the [data] attribute in MtxGridColumn.
Please refer to this [link](https://material.angular.io/components/table/examples#table-filtering) to get a better understanding of the point I'm trying to convey.
I am attaching the screenshot of the error I am receiving.
![mtxgrid](https://github.com/ng-matero/ng-matero/assets/37538741/b5886c8a-b25f-4d15-8369-096c9ef78641)

**Comment by rahulpd101** [comment]
@nzbin
Please suggest a way that is possible via ng-matero.

**Comment by nzbin** [comment]
Your `dataList` should look like the following type
```ts
dataList = [
  {
    position: 1,
    name: 'Boron',
    weight: 10.811,
    symbol: 'B',
    gender: 'male',
    ...
  },
  {
    position: 2,
    name: 'Helium',
    weight: 8.0026,
    symbol: 'He',
    gender: 'female',
  }
  ...
]
```

**Comment by ramax495** [comment]
Are there any plans to support MatTableDataSource in ng-matero ?
